webapp/lib/processors.py

```diff
@@ -181,7 +179,13 @@ def _pilbox_operation(self, operation):
                 if not resize_height and max_height:
                     if max_height < image_info.height:
                         resize_height = max_height
-
+                # conserve the image ratio
+                if resize_height and not resize_width:
+                    image_ratio = image_info.height / resize_height
```

Comment:
Maintain the ration when there is only one of width and height present, otherwise the returned image will look weird